lib/Logger.js

```diff
@@ -32,20 +36,28 @@ class Logger {
      *                                      the 'level' field is not accounted
      *                                      for)
      *
-     * @param {string} name  - The name of the Logger. It can be found later on
-     *                         in the log entries.
      *
      * @see [Bunyan's documentation]{@link
      * https://github.com/trentm/node-bunyan/blob/master/README.md#streams} for
      * a more detailed description of the streams array configuration.
      *
      * @returns {undefined}
      */
-    constructor(config, name) {
-        this.config = config;
+    constructor(name, config) {
+        this.config = new Config();
```

Comment:
Doing this here means that all loggers will have separate config instances; preventing any easy way to do a global logging-level update; which is an ongoin work on MEtadata. Let me explain a use-case:
 * Customer A has a problem
 * Customer A sends info to TS/Devs but there is not enough info
 * Since the problem may be a transient bug that would be fixed by restarting the component, TS asks Customer A to change the loglevel dynamically (REST Admin API for Metadata, for instance); and re-send some logs later
 * Logs a retrieved, provide more information relative to the problem, and the investigation can go on on the dev side.

Essentially, if we create one config instance per module logger; not only this becomes very complex/tedious to do at the level of one process, but we also loose all properties of deduplication of information. I'd advise to stick with what I provided earlier.